src/api/github/__mocks__/octokitWithRetries.ts

```diff
@@ -0,0 +1,183 @@
+import MockCompareCommits from '@test/compareCommits.json';
+import { workflow_run_job } from '@test/payloads/github/workflow_run_job';
+import { MockOctokitError } from './mockError';
+
+
+// Looks goofy, don't it? This is what I could figure out to mock a class. I
+// tried a lot of stuff following these docs and couldn't figure it out other
+// than this:
+//
+//   https://jestjs.io/docs/27.x/es6-class-mocks
+//   https://jestjs.io/docs/27.x/mock-function-api#jestmockedclass
+
+export const OctokitWithRetries = jest.fn(() => {
+  return OctokitWithRetries;
+});
+
+
+OctokitWithRetries.actions = {
+  listWorkflowRunsForRepo: jest.fn(),
+  cancelWorkflowRun: jest.fn(),
+  getJobForWorkflowRun: jest.fn(async ({ job_id }) => {
+    return { data: workflow_run_job({ job_id }) };
+  }),
+};
+
+OctokitWithRetries.apps = {
+  getOrgInstallation: jest.fn(async ({ org }) => {
+    return {
+      data: {
+        id: `installation-${org}`
+      }
+    };
+  }),
+};
+
+OctokitWithRetries.checks = {
+  listForRef: jest.fn(),
+  listAnnotations: jest.fn(async () => {
+    return {
+      data: [
+        {
+          path: 'tests/snuba/rules/conditions/test_event_frequency.py',
+          blob_href:
+            'https://github.com/getsentry/sentry/blob/83ef9b927cbb822febbdf75e5e05dd40afb187cf/tests/snuba/rules/conditions/test_event_frequency.py',
+          start_line: 570,
+          start_column: null,
+          end_line: 570,
+          end_column: null,
+          annotation_level: 'failure',
+          title:
+            'tests/snuba/rules/conditions/test_event_frequency.py#L570',
+          message:
+            'EventFrequencyPercentConditionTestCase.test_one_hour_with_events\n' +
+            '\n' +
+            'AssertionError',
+          raw_details: null,
+        },
+        {
+          path: '.github',
+          blob_href:
+            'https://github.com/getsentry/sentry/blob/83ef9b927cbb822febbdf75e5e05dd40afb187cf/.github',
+          start_line: 1,
+          start_column: null,
+          end_line: 1,
+          end_column: null,
+          annotation_level: 'failure',
+          title: '.github#L1',
+          message: 'Process completed with exit code 2.',
+          raw_details: null,
+        },
+      ],
+    };
+  }),
+};
+
+OctokitWithRetries.git = {
+  getCommit: jest.fn(),
+};
+
+OctokitWithRetries.graphql = jest.fn();
+
+OctokitWithRetries.issues = {
+  _labels: new Set([]),
+  _comments: [],
+  addLabels: jest.fn(async function (payload) {
+    for (const name of payload.labels) {
+      this._labels.add(name);
+    }
+  }),
+  removeLabel: jest.fn(async function (payload) {
+    this._labels.delete(payload.name);
+  }),
+  createComment: jest.fn(async function (payload) {
+    this._comments.push(payload.body);
+  }),
+  update: jest.fn(),
+};
+
+OctokitWithRetries.orgs = {
+  checkMembershipForUser: jest.fn(async function ({ org, username }) {
```

Comment:
I took an opportunity to clean up the call signature here a bit, I had been using `(x) ... x.org` previously. Besides the `apps` addition above, this is the only other material change to the mock client API, sorry I don't have a clean diff for you.